zip-wishing-well.rst

```diff
@@ -0,0 +1,123 @@
+#. Preserve the `Security Properties` below.
+
+Security Properties
+-------------------
+
+#. Wishing-Well-aware applications or users can `verify` a seed to reasonably convince themselves that the ZEC sent into the Well is not recoverable. By implication: it is not possible for anyone to create any typical SIVK prerequisite and also a Well seed for the same SIVK [#].
+#. Any Sapling-enabled wallets or other products can interact with an instance's SIVK and/or address with no awareness of, or specialized features for, the Wishing Well protocol.
+#. It is not possible, without knowledge of either a Well seed or one typical Sapling address prerequisites, to determine whether or not a Sapling address is a Well address, a typical address, or perhaps derived from some other protocol.
+#. It is not possible for a sender to a Well instance to determine whether or not their transfer to a Sapling address is into a Well instance versus a typical address or other case without knowledge of the Well seed.
+#. All privacy properties of Sapling hold against all adversary models and capabilities that apply to typical Sapling usage.
+#. It is not possible for an adversary observing the blockchain to distinguish Wishing Well transactions from other Sapling transactions. Caveats:
+  - Different applications may have different usage patterns, especially timing, which may leave a signature on-chain. If a particular application which relies on the Wishing Well Protocol has a unique detectable usage pattern, this does not impact the on-chain privacy of different applications which rely on the Wishing Well Protocol.
+#. It is not possible for a privacy adversary with knowledge of the Viewing Key to distinguish if that Viewing Key and associated Address is a Well instance versus a typical Sapling address. Caveats:
+  - An adversary who can observe the lack of outgoing transactions may deduce that the associated address is a well instance.
+
+.. [#] This follows from collision resistance of `BLAKE2s-256`. See [FIXME protocol reference].
+
+Specification
+=============
+
+Well Seed Generation
+--------------------
+
+A `Well Seed` is an arbitrary application specific value. However, application designers should be aware of the security properties of their choice. For many applications, a seed should be generated as a 256-bit-entropy value. This ensures the seed is unguessable and will not collide with any other application instances.
+
+An application may want a "globally public" Wishing Well, in which case it may use a well-known constant value.
+
+In particular it is recommended against allowing users to select bare seeds.
+
+Wishing Well Derivation Scheme
+------------------------------
+
+A Sapling incoming viewing key is derived from a `Well Seed` by using `blake2b` with the personalization ASCII string ``zcash-wishing-well-derivation-v1-u8eQd6IG7wsxlr+PC2UFa1GxttqAA159m+tdV3Tl3XI=``. [#]
```

Comment:
This is the first time I've seen a personalization string which contains fixed text and an author-provided random value. While I'm not disputing the security of the construction, I think more context could be helpful.

Is this a standard construction?
If so, can you add a reference to another well-known protocol which uses it?

Is this a necessary construction?
How likely is it that another protocol will use a string which contains those exact 4 words?
Can we avoid any overlap by also specifying the function of the protocol?
(That is, burning, no-spend, no-output.)

Base64 comes with a usability cost - in some fonts, it's difficult for developers to distinguish 1, l, and I.
So they could accidentally create an incompatible protocol.
If we keep the random value, would base32 or hex be easier to type?

Reply:
I'm not really attached to adding a random value to the personalization, which isn't well motivated. Agree it's simpler just to do the standard thing.